File: application/akka/java/AkkaQuickstart.java
package example;

import akka.actor.typed.ActorSystem;

import java.io.IOException;
public class AkkaQuickstart {
  public static void main(String[] args) {
    //#actor-system
    final ActorSystem<GreeterMain.SayHello> greeterMain = ActorSystem.create(GreeterMain.create(), "helloakka");
    //#actor-system

    if (System.getenv("DEMO_USERNAME") == null || System.getenv("DEMO_PASSWORD") == null) {
    	System.out.println("not all variables are set");
    } else {
    	//#main-send-messages
    	greeterMain.tell(new GreeterMain.SayHello(System.getenv("DEMO_USERNAME")));
    	//#main-send-messages
    }
    greeterMain.terminate();
  }
}

File: application/akka/scala/AkkaQuickstart.scala
//#full-example
package example


import akka.actor.typed.ActorRef
import akka.actor.typed.ActorSystem
import akka.actor.typed.Behavior
import akka.actor.typed.scaladsl.Behaviors

//#main-class
object AkkaQuickstart extends App {

  if (sys.env.get("DEMO_USERNAME").getOrElse("").isEmpty() || sys.env.get("DEMO_PASSWORD").getOrElse("").isEmpty()) {
  	println("not all variables are set")
  } else {
  	//#main-send-messages
  	println("Hello " + sys.env.get("DEMO_USERNAME").get)
  	//#main-send-messages
  }
}
//#main-class
//#full-example

